### PR TITLE
add config for event

### PR DESCRIPTION
**Pull request by kaka89**

### CODE REVIEW

File: deploy/k8s.yaml [hunk 1]
```diff
@@ -26,6 +26,12 @@ items:
             value: jdbc:postgresql://reactivesw-data/inventory
           - name: spring_cloud_config_uri
             value: http://reactivesw-config
+          - name: io_reactivesw_message_topic_inventoryreserved_name
+            value: reactivesw-dev-inventory-reserved
+          - name: io_reactivesw_message_topic_inventoryreserved_version
+            value: "1"
+          - name: io_reactivesw_message_topic_ordercreated_subscriber
+            value: dev-payment-order-pay
           - name: spring_datasource_username
```
Comment (davis999) [inline deploy/k8s.yaml:35]: why not add those config to config file on github?

File: src/main/java/io/reactivesw/inventory/application/model/mapper/EventMessageMapper.java [hunk 2]
```diff
@@ -14,28 +14,28 @@
...
    * Build event message.
    *
    * @param orderId the order id
-   * @param status the status
+   * @param status  the status
    * @return the event message
    */
-  public static EventMessage build(String orderId, boolean status) {
+  public static EventMessage build(String orderId, boolean status, EventConfig eventConfig) {
     EventMessage message = new EventMessage();
```
Comment (davis999) [inline src/main/java/io/reactivesw/inventory/application/model/mapper/EventMessageMapper.java:33]: I think a String about topic is fine, do not need the whole EventConfig.

File: src/main/java/io/reactivesw/inventory/domain/service/EventMessageService.java [hunk 3]
```diff
@@ -32,24 +31,30 @@
...
    * Save reserved order event.
    *
    * @param orderId the order id
-   * @param status the status
+   * @param status  the status
    */
   public void saveReservedOrderEvent(String orderId, Boolean status) {
     LOG.debug("Enter. OrderId: {}, status: {}.", orderId, status);
-    EventMessage message = EventMessageMapper.build(orderId, status);
+    EventMessage message = EventMessageMapper.build(orderId, status, eventConfig);
```
Comment (davis999) [inline src/main/java/io/reactivesw/inventory/domain/service/EventMessageService.java:60]: I think topic name is fine, do not need the whole EventConfig.

File: src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java [hunk 4]
```diff
@@ -6,15 +6,33 @@
...
+   * Inventory reserved topic version.
+   */
+  @Value("${io.reactivesw.message.topic.inventoryreserved.version}")
+  private Integer inventoryReservedVersion;
+
+  /**
+   * Subscriber of order created.
+   */
+  @Value("${io.reactivesw.message.topic.ordercreated.subscriber}")
+  private String orderCreatedSubscriber;
```
Comment (davis999) [inline src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java:37]: orderCreationSubscriber will be better.

File: src/main/java/io/reactivesw/inventory/domain/service/EventMessageService.java [hunk 5]
```diff
@@ -32,24 +31,30 @@
...
+   * Event message config.
+   */
+  private transient EventConfig eventConfig;
+
+  /**
    * Instantiates a new Event message service.
    *
    * @param eventMessageRepository the event message repository
    */
-  @Autowired
```
Comment (davis999) [inline src/main/java/io/reactivesw/inventory/domain/service/EventMessageService.java]: can remove @Autowired?

File: src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java [hunk 6]
```diff
@@ -6,15 +6,33 @@
...
+   */
+  @Value("${io.reactivesw.message.topic.inventoryreserved.name}")
+  private String inventoryReservedName;
+
+  /**
+   * Inventory reserved topic version.
+   */
+  @Value("${io.reactivesw.message.topic.inventoryreserved.version}")
+  private Integer inventoryReservedVersion;
+
```
Comment (davis999) [inline src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java:32]: if this service has other event message, you have to change this name or add a new version? 
eventVersion will be better?

File: src/main/java/io/reactivesw/inventory/application/model/mapper/EventMessageMapper.java (hunk 2 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/inventory/application/model/mapper/EventMessageMapper.java:33]: we need topic, version

File: src/main/java/io/reactivesw/inventory/domain/service/EventMessageService.java (hunk 3 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/inventory/domain/service/EventMessageService.java:60]: we need topic, version

File: src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java (hunk 6 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java:32]: each topic should has it's own version

File: src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java (hunk 4 above)
Comment (kaka89) [reply src/main/java/io/reactivesw/inventory/infrastructure/configuration/EventConfig.java:37]: keep the name with the topic will be ok

File: deploy/k8s.yaml (hunk 1 above)
Comment (kaka89) [reply deploy/k8s.yaml:35]: yes, we need to add these to config file.

**Pull request merged**